cranelift-codegen/src/regalloc/branch_splitting.rs

```diff
@@ -14,31 +14,33 @@ use crate::ir::{Ebb, Function, Inst, InstBuilder, InstructionData, Opcode, Value
 use crate::isa::TargetIsa;
 use crate::topo_order::TopoOrder;
 
+/// Split possible critical edges. Returns a list of added blocks.
 pub fn run(
-    isa: &TargetIsa,
+    isa: &dyn TargetIsa,
     func: &mut Function,
     cfg: &mut ControlFlowGraph,
     domtree: &mut DominatorTree,
     topo: &mut TopoOrder,
-) {
+) -> Vec<Ebb> {
```

Comment:
Looking at the rest of the Cranelift phases, usually it seems that these values are provided at a `&mut` as argument of the run command instead of returned from the `run` function. Then the `clear` function of the register allocator is used to clean the content of the vector, to be reused without reallocation for the next compiled function.